common/lib/xmodule/xmodule/modulestore/xml_importer.py

```diff
@@ -753,15 +753,26 @@ def _convert_ref_fields_to_new_namespace(reference):  # pylint: disable=invalid-
         # if library exists, update source_library_version and children
         # according to this existing library and library content block.
         if store.get_library(block.source_library_key):
-            LibraryToolsService(store).update_children(
-                block,
-                user_id,
-                version=block.source_library_version
-            )
+            update_block_children_on_branch(store, block, user_id, store.get_branch_setting())
 
     return block
 
 
+def update_block_children_on_branch(store, lc_block, user_id, branch):
+    """
+    Updates library content block's children on draft and published branches.
+    """
+    with store.branch_setting(branch_setting=branch):
```

Comment:
Hey @cpennington I need help here, this code is supposed to update randomize content block children:
1. for both `published `and `draft` branches in case if store's branch setting has been set to `published-only` (i.e. when we are importing published blocks). 
2. for only `draft` branch in case if store's branch setting has been set to `draft-preferred` (i.e. when we are importing from draft directory).

But while debugging I've seen that `store.get_branch_setting()` is always returning `draft-preferred` no matter what branch has been set in context manager i.e. `with self.store.branch_setting("published-only"):`

Like in [here](https://github.com/edx/edx-platform/blob/c1366651d4f6341783afd514a5aee820874bca3f/common/lib/xmodule/xmodule/modulestore/xml_importer.py#L558), all the method calls in this manager should have store with branch setting `published-only` but I'm getting `draft-preferred` when I use `store.get_branch_setting()` in my method. I am confused here. Actually, I need branch setting for this method to work for appropriate branch(es).